In [1]:
from faker import Faker
import Dict
import NamedTuple

In [2]:
profile_named_tuple = NamedTuple.form_profile_named_tuple()

#profile_dict = Dict.form_profile_dict()
profile_dict = Dict.form_profile_dict_from_namedtuple(profile_named_tuple)

In [3]:
# largest blood type
dict_largest_blood_type = Dict.calculate_largest_blood_type(profile_dict)
namedtuple_largest_blood_type = NamedTuple.calculate_largest_blood_type(profile_named_tuple)
print('largest blood type and time taken by namedtuple is ', namedtuple_largest_blood_type)
print('largest blood type and time taken time taken by dict is ', dict_largest_blood_type)

largest blood type and time taken by namedtuple is  ('B-', 0.003495931625366211)
largest blood type and time taken time taken by dict is  ('B-', 0.00608515739440918)


In [9]:
# mean current location
dict_mean_current_location = Dict.calculate_mean_current_location(profile_dict)
namedtuple_mean_current_location = NamedTuple.calculate_mean_current_location(profile_named_tuple)
print('mean current location and time taken by namedtuple is ', namedtuple_mean_current_location)
print('mean current location and time taken time taken by dict is ', dict_mean_current_location)

mean current location and time taken by namedtuple is  ((Decimal('0.1534287727'), Decimal('1.126247345')), 0.008796215057373047)
mean current location and time taken time taken by dict is  ((Decimal('0.1534287727'), Decimal('1.126247345')), 0.009984016418457031)


In [10]:
# oldest person age
dict_oldest_person_age = Dict.calculate_oldest_person_age(profile_dict)
namedtuple_oldest_person_age = NamedTuple.calculate_oldest_person_age(profile_named_tuple)
print('oldest person age and time taken by namedtuple is ', namedtuple_oldest_person_age)
print('oldest person age and time taken time taken by dict is ', dict_oldest_person_age)

oldest person age and time taken by namedtuple is  (115.96167008898016, 0.02734375)
oldest person age and time taken time taken by dict is  (115.96167008898016, 0.03199577331542969)


In [7]:
# Average age
dict_avg_age = Dict.calculate_avg_age(profile_dict)
namedtuple_avg_age = NamedTuple.calculate_avg_age(profile_named_tuple)

print('Average person age and time taken by namedtuple is ', namedtuple_avg_age)
print('Average person age and time taken time taken by dict is ', dict_avg_age)

Average person age and time taken by namedtuple is  (57.59003394934976, 0.020982027053833008)
Average person age and time taken time taken by dict is  (57.59003394934976, 0.024750947952270508)


# Conclusion
## From the logs above the time taken by namedtuple is less than dict
| Task | Time taken by namedtuple | Time taken by dict |
| --- | --- | --- |
| Largest Blood group | 0.003495931625366211 | 0.00608515739440918 |
| Mean current location | 0.008796215057373047 | 0.009984016418457031 |
| Oldest person age | 0.02734375 | 0.03199577331542969 |
| Average age | 0.020982027053833008 | .0.024750947952270508 |
